transformations/multilingual_dictionary_based_code_switch/transformation.py

```diff
@@ -88,11 +84,21 @@ def __init__(self, seed=0, max_outputs=1, code_switch_rate=0.9):
         self.switch_dict = load_dict()
 
     def generate(self, sentence: str):
-        words = sentence.split(" ")
-        out = ""
-        for word in words:
-            out += code_switch(word, self.switch_dict, self.code_switch_rate, self.seed)
-        return [out]
+        random.seed(self.seed)
+        ouputs = []
+        i = 0
+        while i < self.max_outputs:

```

Comment:
There's a danger the while loop will not terminate if the space of possible outputs is smaller than `max_outputs`. Since it is `max_outputs`, perhaps you can either use `num_tries` variable to track the number of tries and terminate once it reaches a certain value regardless of the actual outputs generated, or just use `range(max_outputs)`.

Reply:
@samsontmr Got it.  fixed~